syntax = "proto3";

package teste;

message Movie {
  optional string id = 1;
  string title = 2;
  repeated string cast = 3;
  repeated string genres = 4;
  int32 runtime = 5;
  int32 year = 6;
  string type = 7;
}

message MoviesList {
  repeated Movie movies = 1;
}

message Command {
  string message = 1;
}

message Response {
  string message = 1;
}


service MovieService {
  rpc CreateMovie(Movie) returns (Response) {}
  rpc ListMoviesByCast(Command) returns (stream Movie) {}
  rpc ListMoviesByGenres(Command) returns (stream Movie) {}
  rpc UpdateMovie(Movie) returns (Response) {}
  rpc DeleteMovie(Command) returns (Response) {}
}
